test/functionalTest/cases/2015_notification_templates/notification_templates.test

```diff
@@ -0,0 +1,544 @@
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Notification Templates, Normal Case
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Create subscription for E1 with only URL in template part
+# 02. Create subscription for E1 with URL and VERB in template part
+# 03. Create subscription for E1 with URL and PAYLOAD in template part - PAYLOAD with substitutions
+# 04. Create subscription for E1 with URL and QS in template part - QS with substitutions
+# 05. Create subscription for E1 with URL and HEADERS in template part - HEADERS with substitutions
+# 06. Create subscription for E1 with URL, VERB, PAYLOAD, QS and HEADERS in template part - all with substitutions (except VERB, of course)
+# 07. Create subscription for E1 with PAYLOAD, QS and HEADERS with unknown attribute 'A4' for substitution
+# 08. Create E1 with attributes A1, A2 and A3
+# 09. Dump accumulator, see 7 notifications
+#
+
+echo "01. Create subscription for E1 with only URL in template part"
+echo "============================================================="
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id" : "E1"
+      }
+    ],
+    "condition": {
+      "attrs": []
+    }
+  },
+  "notification": {
+    "httpExtended": {
+      "url":     "http://127.0.0.1:'${LISTENER_PORT}'/${NOTIFY_URL}",
+      "qs":      { "step": "01" }
+    }
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "02. Create subscription for E1 with URL and VERB in template part"
+echo "================================================================="
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id" : "E1"
+      }
+    ],
+    "condition": {
+      "attrs": []
+    }
+  },
+  "notification": {
+    "httpExtended": {
+      "url":     "http://127.0.0.1:'${LISTENER_PORT}'/${NOTIFY_URL}",
+      "method":  "PATCH",
+      "qs":      { "step": "02" }
+    }
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "03. Create subscription for E1 with URL and PAYLOAD in template part - PAYLOAD with substitutions"
+echo "================================================================================================="
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id" : "E1"
+      }
+    ],
+    "condition": {
+      "attrs": []
+    }
+  },
+  "notification": {
+    "httpExtended": {
+      "url":     "http://127.0.0.1:'${LISTENER_PORT}'/notify",
+      "payload": "{ \"A1\": \"${A1}\", \"A2\": \"${A2}\", \"A3\": \"${A3}\", \"step\": \"03\" }",
```

Comment:
Not sure what fields you are referring to.
Here it seems we need to start checking forbidden chars before we can do what you ask.